src/adapter/src/rbac.rs

```diff
@@ -986,6 +990,13 @@ fn generate_required_privileges(
             }
             ObjectId::Cluster(_) | ObjectId::Database(_) | ObjectId::Role(_) => Vec::new(),
         },
+        Plan::AlterSetInCluster(plan::AlterSetInClusterPlan { id: _, set_cluster }) => {
```

Comment:
This should also require DELETE of the old cluster.

Reply:
Is this only required for linked clusters? I think we can distinguish the following cases:
* The materialized view is part of a regular cluster. It can be created by a user that has `CREATE` privileges on the qualifier and cluster. It can be dropped by a user that has `USAGE` access on the qualifier (and owns the object).
* Linked clusters are not special-cased for RBAC, but need to be dropped when moving an item away from the linked cluster, so we need to have the same privileges that `DropItem` would require. This is cluster `USAGE` (and ownership I presume?), nothing else.

With this reasoning, I'm a bit at a loss as to what the privileges should be, besides `CREATE` on the new cluster. Also, I think that sequencing a `ALTER SOURCE ... SET IN CLUSTER` with a linked cluster results in `[AlterSetInCluster, DropClusterReplica, DropCluster]`, which should require the privileges mentioned above.